bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigResolver.java

```diff
@@ -65,9 +68,15 @@ public ConfigResolver(Map<Module, VariableKey[]> configVarMap,
         }
         for (Map.Entry<Module, VariableKey[]> entry : configVarMap.entrySet()) {
             Module module = entry.getKey();
+            String moduleName = module.getName();
             VariableKey[] variableKeys = entry.getValue();
             for (VariableKey varKey : variableKeys) {
                 Optional<?> configValue = getConfigValue(module, varKey, runtimeConfigProviders);
+                if (varKey.isRequired && configValue.isEmpty()) {
+                    String configVarName =
+                            (moduleName.equals(DEFAULT_MODULE)) ? varKey.variable : moduleName + ":" + varKey.variable;
+                    throw new ConfigException(String.format(VALUE_NOT_PROVIDED, configVarName));
```

Comment:
AFAIR  earlier we had codegen to handle value not provided. Maybe we can remove that. Let's do that after merging this PR if needed.

Reply:
I think it ok to leave the other check as well, just in case. We can change the message.